components/hab/src/main.rs

```diff
@@ -841,6 +843,51 @@ fn sub_pkg_sign(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     command::pkg::sign::start(ui, &pair, &src, &dst)
 }
 
+fn sub_pkg_bulkupload(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
+    let upload_dir = bulkupload_dir_from_matches(m);
+    // upload_dir is required via clap and exists
+    let artifact_path = upload_dir.as_ref().unwrap().join("artifacts");
+    let key_path = upload_dir.as_ref().unwrap().join("keys");
+    let url = bldr_url_from_matches(&m)?;
+    let additional_release_channel = channel_from_matches(&m);
+    let force_upload = m.is_present("FORCE");
+    let auto_build = if m.is_present("AUTO_BUILD") {
+        BuildOnUpload::PackageDefault
+    } else {
+        BuildOnUpload::Disable
+    };
+    let token = auth_token_param_or_env(&m)?;
+    let options = glob::MatchOptions { case_sensitive:              true,
+                                       require_literal_separator:   true,
+                                       require_literal_leading_dot: true, };
+    let artifact_paths =
+        vec_from_glob_with(&artifact_path.join("*.hart").display().to_string(), options);
+
+    ui.begin(format!("Preparing to upload artifacts to the '{}' channel on {}",
+                     additional_release_channel.clone()
+                                               .unwrap_or_else(ChannelIdent::unstable),
```

Comment:
Is defaulting to the unstable channel an inherent assumption of the builder API? I worry about the fact that this could become misleading if that were to ever change and there's nothing that ties the modification of the two together. Perhaps it would be safer to require the channel specification to be explicit.

Reply:
I don't think that would ever change though. From my understanding, `unstable` is designed as the default birthing channel for all artifacts and the one in which they all must reside even if no other.